question-servers/freeformPythonLib/prairielearn.py

```diff
@@ -17,6 +18,31 @@
 import collections
 
 
+# TODO in the future it would be nice to have a way to use the `Final` type
+# around some of these values for the different functions so that the type
+# checker knows which functions are allowed to modify which values.
+# This may require subclassing QuestionData, and constructing the `data`
+# dictionary through class constructors instead of the usual was
```

Comment:
Is this comment out of date given "it seems like types of properties can't be redefined in subclasses, even if the new type is a subtype of the type in the parent class"?

Reply:
Ya, fixed.